syntax = "proto3";

message BlockHeader {
    fixed32 entropy = 1;
    double timestamp = 2;
    fixed32 difficulty = 3;
    bytes body_hash = 4;
}

message BlockBody {
    repeated bytes blobs = 1;
}

message Block {
    fixed32 nonce = 1;
    BlockHeader header = 2;
    BlockBody body = 3;
}
